libbpf-cargo/src/gen/btf.rs

```diff
@@ -0,0 +1,627 @@
+                }
+            }
+            BtfKind::Float(t) => {
+                let width = match t.size() {
+                    2 => bail!("Unsupported float width"),
+                    4 => "32",
+                    8 => "64",
+                    12 => bail!("Unsupported float width"),
+                    16 => bail!("Unsupported float width"),
+                    _ => bail!("Invalid float width"),
+                };
+
+                format!("f{width}")
+            }
+            BtfKind::Ptr(t) => {
+                let pointee_ty = self.type_declaration(t.referenced_type())?;
+
+                format!("*mut {pointee_ty}")
+            }
+            BtfKind::Array(t) => {
+                let val_ty = self.type_declaration(t.contained_type())?;
+
+                format!("[{}; {}]", val_ty, t.capacity())
+            }
+            BtfKind::Struct => self.get_type_name_handling_anon_types(&ty).into_owned(),
+            BtfKind::Union => self.get_type_name_handling_anon_types(&ty).into_owned(),
+            BtfKind::Enum => self.get_type_name_handling_anon_types(&ty).into_owned(),
+            BtfKind::Enum64 => self.get_type_name_handling_anon_types(&ty).into_owned(),
+            //    // The only way a variable references a function is through a function pointer.
+            //    // Return c_void here so the final def will look like `*mut c_void`.
+            //    //
+            //    // It's not like rust code can call a function inside a bpf prog either so we don't
+            //    // really need a full definition. `void *` is totally sufficient for sharing a pointer.
+            BtfKind::Func => "std::ffi::c_void".to_string(),
+            BtfKind::Var(t) => self.type_declaration(t.referenced_type())?,
+            BtfKind::Fwd => bail!("Invalid type: {ty:?}"),
+            BtfKind::FuncProto => bail!("Invalid type: {ty:?}"),
+            BtfKind::DataSec => bail!("Invalid type: {ty:?}"),
+            BtfKind::Typedef => bail!("Invalid type: {ty:?}"),
+            BtfKind::Volatile => bail!("Invalid type: {ty:?}"),
+            BtfKind::Const => bail!("Invalid type: {ty:?}"),
+            BtfKind::Restrict => bail!("Invalid type: {ty:?}"),
+            BtfKind::DeclTag => bail!("Invalid type: {ty:?}"),
+            BtfKind::TypeTag => bail!("Invalid type: {ty:?}"),
+        });
+        Ok(s)
+    }
+
+    /// Returns an expression that evaluates to the Default value
+    /// of a type(typeid) in string form.
+    ///
+    /// To be used when creating a impl Default for a structure
+    ///
+    /// Rule of thumb is `ty` must be a type a variable can have.
+    ///
+    /// Type qualifiers are discarded (eg `const`, `volatile`, etc).
+    fn type_default(&self, ty: BtfType<'s>) -> Result<String> {
+        let ty = ty.skip_mods_and_typedefs();
+
+        Ok(btf_type_match!(match ty {
+            BtfKind::Void => "std::ffi::c_void::default()".to_string(),
```

Comment:
yeah, it doesn't make much sense either because you can't have an instance of `void` in C or `c_void` in rust, only behind a pointer. So this code path will never be used. I'll replace with a `bail!` call